server/src/main/java/org/opensearch/index/shard/IndexingStats.java

```diff
@@ -223,8 +381,53 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
             builder.field(Fields.IS_THROTTLED, isThrottled);
             builder.humanReadableField(Fields.THROTTLED_TIME_IN_MILLIS, Fields.THROTTLED_TIME, getThrottleTime());
+
+            if (getDocStatusStats() != null) {
+                getDocStatusStats().toXContent(builder, params);
+            }
+
             return builder;
         }
+
+        @Override
+        public boolean equals(Object o) {
```

Comment:
equals & hashCode was not present before for IndexingStats?

Reply:
No, they weren't implemented earlier. I've added them to improve UT coverage through `IndexingStatsTests` class.

Reply:
I am wondering could that change any behavior internally as equality & hashCode logic would change.

Reply:
Ok. I'll revert the change and implement serializing tests like we have for `NodesStatsTests`.